src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -93,13 +124,20 @@ internal RedisCache(IOptions<RedisCacheOptions> optionsAccessor, ILogger logger)
         _logger = logger;
 
         // This allows partitioning a single backend cache for use with multiple apps/services.
-        _instance = _options.InstanceName ?? string.Empty;
+        var instanceName = _options.InstanceName;
+        if (!string.IsNullOrEmpty(instanceName))
+        {
+            // SE.Redis allows efficient append of key-prefix scenarios, but we can help it
+            // avoid some work/allocations by forcing the key-prefix to be a byte[]; SE.Redis
+            // would do this itself anyway, using UTF8
+            _instancePrefix = Encoding.UTF8.GetBytes(instanceName);
```

Comment:
Is this behavior different from `.WithPrefix()` when getting the `IDatabase`? Would tidy up a bit to switch to the built-in if not.

Reply:
Semantically the same, but: as-written is more efficient in this case; if you look at `KeyspaceIsolation.WrapperBase<TInner>.ToInner(RedisKey[])`, it has to allocate a *copy* of the array each call, and `ScriptEvaluate` always uses `RedisKey[]` here. I wanted to avoid that over-and-over-and-over.

What we could do, moving forward, is add some overloads that take `ReadOnlyMemory<>` for a few key methods that currently take arrays, allowing pool leases to be used instead, but: we don't have that today.